### PR TITLE
[direct task] Fix bug that starts duplicate connections from the worker to the local raylet

### PR DESCRIPTION
**Pull request by stephanie-wang**
## Why are these changes needed?

If a task is spilled back to a remote node, then spilled back again to the local node, then the worker will mistakenly create a second grpc client to the local node. This should be okay in most cases but could cause problems if the local raylet fails, since the worker will not fate-share with the local raylet.

This fixes the bug by checking the retry_at_raylet_id in the spillback message against the local raylet ID.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

**Review by ericl — APPROVED** [review]

File: src/ray/core_worker/core_worker.cc [hunk 1]
```diff
@@ -207,7 +207,7 @@ CoreWorker::CoreWorker(const WorkerType worker_type, const Language language,
             return std::shared_ptr<RayletClient>(
                 new RayletClient(std::move(grpc_client)));
           },
-          memory_store_, task_manager_,
+          memory_store_, task_manager_, raylet_id,
```
Comment (edoakes) [inline src/ray/core_worker/core_worker.cc]: Can we rename these to "local_raylet_id_" and "local_raylet_client_" now that we talk to multiple raylets?

Comment (stephanie-wang) [reply src/ray/core_worker/core_worker.cc]: I don't know if it's really necessary, the CoreWorker doesn't talk to other nodes or raylets directly (it's encapsulated in the transport classes).

Comment (edoakes) [reply src/ray/core_worker/core_worker.cc]: Is there any downside? Seems like it makes things strictly more clear.

**Review by edoakes — APPROVED** [review]
LGTM

**Pull request merged**